tests/tests/test_train_schedule.py

```diff
@@ -1415,6 +1415,65 @@ def test_etcs_stop_at_requirements_eoa(
     assert spacing_req_zone_stop["end_time"] == 842_038
 
 
+def test_etcs_train_schedule_with_margins(
+    etcs_scenario: Scenario, etcs_rolling_stock: int, session: Session
+):
```

Comment:
nit: Adding a little context would be nice when we later come back to it (you may have better idea):
```suggestion
def test_etcs_train_schedule_with_margins(
    etcs_scenario: Scenario, etcs_rolling_stock: int, session: Session
):
    """
    Testing a valid ETCS train-schedule with margin that was failing because of
    a tricky case of braking-curve interpolation
    """
```

Reply:
I'll add it with a slight modification `case of speed interpolation`.